zebrad/tests/acceptance.rs

```diff
@@ -1052,3 +1056,161 @@ async fn tracing_endpoint() -> Result<()> {
 
     Ok(())
 }
+
+/// Test will start 2 zebrad nodes one after the other using the same Zcash listener.
+/// It is expected that the first node spawned will get exclusive use of the port.
+/// The second node will panic with the Zcash listener conflict hint added in #1535.
+#[test]
+fn zcash_listener_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let listen_addr = format!("127.0.0.1:{}", port);
+
+    // Write a configuration that has our created network listen_addr
+    let mut config = default_test_config()?;
+    config.network.listen_addr = listen_addr.parse().unwrap();
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened Zcash protocol endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same listener.
+    // `network.listen_addr` will be the same in the 2 nodes.
+    // (But since the config is ephemeral, they will have different state paths.)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, "(already in use)|(one usage)")?;
+
+    Ok(())
+}
+
+/// Start 2 zebrad nodes using the same metrics listener port, but different
+/// state directories and Zcash listener ports. The first node should get
+/// exclusive use of the port. The second node will panic with the Zcash metrics
+/// conflict hint added in #1535.
+#[test]
+fn zcash_metrics_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let listen_addr = format!("127.0.0.1:{}", port);
+
+    // Write a configuration that has our created metrics endpoint_addr
+    let mut config = default_test_config()?;
+    config.metrics.endpoint_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened metrics endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same endpoint.
+    // `metrics.endpoint_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, "(already in use)|(one usage)")?;
```

Comment:
These regexes are out of sync with the issue_filter string matching code.
Let's create a constant for this regex, and use it in both places.